uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -488,11 +481,16 @@ define([
     function setCalendarProps(contactID, calendar) {
       var dateObj,
         leaveRequest,
-        dates = Object.keys(calendar.days);
+        dates = Object.keys(calendar.days),
+        indexedByMonth = angular.copy(vm.months.map(function () {
+          return [];
+        }));
 
       dates.forEach(function (date) {
         dateObj = calendar.days[date];
-        leaveRequest = getLeaveRequestByDate(contactID, dateObj.date);
+        indexedByMonth[parseInt(moment(dateObj.date).month())].push(dateObj);
+        leaveRequest = leaveRequestsIndexed[contactID] ? leaveRequestsIndexed[contactID][dateObj.date]: null;
+        leaveRequest = (leaveRequest && leaveRequest.contact_id === contactID) ? leaveRequest : null;
```

Comment:
This looks pretty bad, tbh. You are doing some convoluted two-steps assignment that is pretty hard to follow. I don't know if it's a matter of how the `leaveRequestsIndexed` is named, or how it is structured (`leaveRequestsIndexed[contactID][dateObj.date]`), but it doesn't come naturally to follow. Please explain it and let's figure out a way to improve this